emoji/src/main/java/com/gluonhq/emoji/DownloadbleEmojiSpriteLoader.java

```diff
@@ -0,0 +1,104 @@
+    private static final String EMOJI_PNG_URL = "https://github.com/iamcal/emoji-data/blob/" + COMMIT_NUMBER + "/sheets-clean/sheet_apple_%s_clean.png?raw=true";
+
+    private static final String LOCAL_PATH = System.getProperty("user.home") + "/.gluon/emoji/" + COMMIT_NUMBER;
+    private static final int[] EMOJI_SIZES = new int[] { 20, 32, 64 };
+    private boolean initialized;
+
+    @Override
+    public boolean isInitialized() {
+        if (!initialized) {
+            for (int size : EMOJI_SIZES) {
+                String fileName = "sheet_apple_" + size + ".png";
+                if (!Files.exists(Paths.get(LOCAL_PATH, fileName))) {
+                    return false;
+                }
+            }
+        }
+        return initialized = true;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> initialize() {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                downloadSprites(EMOJI_SIZES);
+                initialized = true;
+                return true;
+            } catch (Exception e) {
+                return false;
+            }
+        });
+    }
+
+    private void downloadSprites(int... sizes) {
+        for (int size : sizes) {
+            String fileName = "sheet_apple_" + size + ".png";
+            Path localPath = Paths.get(LOCAL_PATH, fileName);
+            try {
+                if (!Files.exists(localPath)) {
+                    LOG.fine("Download sprite file for size: " + size);
+                    downloadFile(new URI(String.format(EMOJI_PNG_URL, size)).toURL(), localPath);
+                }
+            } catch (IOException | URISyntaxException e) {
+                LOG.severe("Download sprite failed: " + e);
+                throw new RuntimeException("Unable to load local image file", e);
+            }
+        }
+    }
+
+    public Image loadEmojiSprite(int size) {
+        if (!initialized) {
+            throw new RuntimeException("Sprite Loader hasn't been initialized or completed initialization");
+        }
+        String fileName = "sheet_apple_" + size + ".png";
+        Path localPath = Paths.get(LOCAL_PATH, fileName);
+        try {
+            return new Image(new FileInputStream(localPath.toFile()));
+        } catch (IOException e) {
+            LOG.severe("Loading of local image file failed: " + e);
+            throw new RuntimeException("Unable to load local image file", e);
+        }
+    }
+
+    @Override
+    public InputStream loadCSV() {
+        return DownloadbleEmojiSpriteLoader.class.getResourceAsStream("emoji.csv");
+    }
+
+    private void downloadFile(URL url, Path filePath) throws IOException {
+        Path parentDir = filePath.getParent();
+        if (parentDir != null && !Files.exists(parentDir)) {
+            Files.createDirectories(parentDir);
+        }
+        ReadableByteChannel readableByteChannel = Channels.newChannel(url.openStream());
```

Comment:
Why don't you use Files.copy(url.openStream()) here?

Reply:
done!